pkg/controller/appdefinition/pullappimage.go

```diff
@@ -83,8 +83,12 @@ func determineTargetImage(appInstance *v1.AppInstance) (string, string) {
 			}
 		}
 	} else {
-		// Auto-upgrade is off. Only need to pull if spec and status are not equal
+		// Auto-upgrade is off. Only need to pull if spec and status are not equal or we're trying to trigger a repull
 		if appInstance.Spec.Image == appInstance.Status.AppImage.Name {
+			// Repull triggered
+			if appInstance.Status.AvailableAppImage == appInstance.Spec.Image {
+				return appInstance.Spec.Image, ""
+			}
 			return "", ""
 		} else {
 			return appInstance.Spec.Image, ""
```

Comment:
non-blocking: We can flip the if statement here to keep the code a bit dryer and more fitting for the comment.
```suggestion
		// Auto-upgrade is off. Only need to pull if spec and status are not equal or we're trying to trigger a repull
		if appInstance.Spec.Image != appInstance.Status.AppImage.Name || 
		appInstance.Status.AvailableAppImage == appInstance.Spec.Image {
		        return appInstance.Spec.Image, ""
		} else {
		        return "", ""
```

Reply:
makes sense. thanks!